test/distributed/test_c10d_gloo.py

```diff
@@ -1738,48 +1731,6 @@ def forward(self, x):
             loss = criterion(output, target)
             loss.backward()
 
-    @requires_gloo()
-    @skip_if_lt_x_gpu(2)
-    def test_ignored_sharded_tensor(self):
-        class MyModule(nn.Module):
-            def __init__(self, shard_tensor: ShardedTensor) -> None:
-                super().__init__()
-                self.fc1 = nn.Linear(2, 10, bias=False)
-                self.st = nn.Parameter(shard_tensor)
-                self.relu = nn.ReLU()
-
-            def forward(self, x):
-                x = self.relu(self.fc1(x))
-                return F.softmax(x, dim=1)
-        pg = dist.init_process_group(
-            "gloo",
-            init_method=f"file://{self.file_name}",
-            world_size=self.world_size,
-            rank=self.rank,
-        )
-        device = torch.device(f"cuda:{self.rank}")
-        local_shard_metadata = ShardMetadata(
-            shard_offsets=[(self.rank % 2) * 5, 0],
-            shard_sizes=[5, 10],
-            placement=f"rank:{self.rank}/cuda:{self.rank}"
-        )
-        local_shards = [Shard(torch.randn(5, 10, device=device), local_shard_metadata)]
-        st = init_from_local_shards(local_shards, [10, 10])
-        m = MyModule(st)
-        with _ddp_replicated_tensor(False):
```

Comment:
same here, shouldn't we just remove the replicated tensor disable, and ensure that DDP still ignores Shardedtensors?

Reply:
ok, since we are deprecating ST anyway. I thought this is something we can also remove.. I will only remove the replicated tensor part for now.